Products/CMFPlone/browser/favicon.py

```diff
@@ -1,26 +1,44 @@
+from os.path import dirname, join
+
 from Products.CMFPlone.interfaces import ISiteSchema
 from plone.formwidget.namedfile.converter import b64decode_file
-from plone.namedfile.browser import Download
+from plone.memoize import ram
+from plone.namedfile.browser import DisplayFile
 from plone.namedfile.file import NamedImage
 from plone.registry.interfaces import IRegistry
 from zope.component import getUtility
-from plone.memoize import ram
-
 
-class SiteFavicon(Download):
 
+class SiteFavicon(DisplayFile):
     def __init__(self, context, request):
         super().__init__(context, request)
         self.filename = None
```

Comment:
Minor: given that you are assigning `self.filename` below anyway there is no need for this line